lib/torrent.js

```diff
@@ -1029,36 +1033,38 @@ export default class Torrent extends EventEmitter {
 
     this._debug('select %s-%s (priority %s)', start, end, priority)
 
-    this._selections.push({
+    this._selections.insert({
       from: start,
       to: end,
       offset: 0,
       priority,
-      notify: notify || noop
+      notify,
+      isStreamSelection
     })
 
     this._selections.sort((a, b) => b.priority - a.priority)
 
     this._updateSelections()
   }
 
-  deselect (start, end, priority) {
+  select (start, end, priority, notify = noop) {
+    this._select(start, end, priority, notify, false)
+  }
+
+  _deselect (start, end, isStreamSelection = false) {
     if (this.destroyed) throw new Error('torrent is destroyed')
 
-    priority = Number(priority) || 0
-    this._debug('deselect %s-%s (priority %s)', start, end, priority)
+    this._debug('deselect %s-%s', start, end)
 
-    for (let i = 0; i < this._selections.length; ++i) {
-      const s = this._selections[i]
-      if (s.from === start && s.to === end && s.priority === priority) {
-        this._selections.splice(i, 1)
-        break
-      }
-    }
+    this._selections.remove({ from: start, to: end, isStreamSelection })
```

Comment:
:/ this throws typeerror during static analysis
```
Argument of type '{ from: any; to: any; isStreamSelection: boolean; }' is not assignable to parameter of type 'SelectionItem'.
  Type '{ from: any; to: any; isStreamSelection: boolean; }' is missing the following properties from type '{ offset: number; priority: number; notify: Function; isStreamSelection?: boolean | undefined; }': offset, priority, notifyts(2345)
```

Reply:
Thanks, I updated it to 
```js
/**
* @param {MinimalSelectionItem & {isStreamSelection: boolean, priority?: number}} item 
*/